src/store.ts

```diff
@@ -153,17 +175,43 @@ export default class Store {
    * Initializes the database by destroying the existing one if needed.
    */
   public async init(resetDB = false): Promise<void> {
-    if (this.db) {
-      await this.dbDestroy()
-    }
+    if (resetDB || !this.db) {
```

Comment:
I changed the logic of how we reset the db. Without this, in tests `init` would get called once on creating the `memoryStore` and again in `start`, which would clear everything out. This makes it require a flag to actually destroy the db.